supportedBackends/spark/src/main/scala/cromwell/backend/impl/spark/SparkClusterProcess.scala

```diff
@@ -0,0 +1,151 @@
+import org.slf4j.LoggerFactory
+import spray.httpx.unmarshalling._
+import scala.util.{Failure, Success, Try}
+
+object SparkClusterProcess {
+
+  case class SparkDriverStateQueryResponse(action: String, driverState: String, serverSparkVersion: String,
+                                           submissionId: String, success: Boolean, workerHostPort: String, workerId: String)
+
+  case class SparkJobSubmissionResponse(action: String, message: String, serverSparkVersion: String, submissionId: String, success: Boolean)
+
+  sealed trait TerminalStatus
+  case class Failed(error: Throwable) extends TerminalStatus
+  case object Finished extends TerminalStatus
+
+  object SparkClusterJsonProtocol extends DefaultJsonProtocol {
+    implicit val sparkStatusResponseFormat = jsonFormat7(SparkDriverStateQueryResponse)
+    implicit val successfulParsedResponseFormat = jsonFormat5(SparkJobSubmissionResponse)
+  }
+}
+
+trait SparkClusterRestClient {
+  def sendAndReceive: SendReceive
+  def makeHttpRequest(httpRequest: HttpRequest): Future[HttpResponse]
+}
+
+trait SparkClusterProcessMonitor {
+  def startMonitoringSparkClusterJob(jobPath: Path,  jsonFile: String): Future[TerminalStatus]
+  def monitorSparkClusterJob(subId: String, rcPath: Path, promise: Promise[Unit]): Unit
+  def evaluateMonitoringFuture(rcPath: Path): Unit
+  def completeMonitoringProcess(rcPath: Path, status: String, promise: Promise[Unit]): Unit
+}
+
+trait SparkClusterJobParser {
+  def parseJsonForSubmissionIdAndStatus(jsonFile: Path): SparkJobSubmissionResponse
+}
+
+class SparkClusterProcess(implicit system: ActorSystem) extends SparkProcess
+  with SparkClusterRestClient with SparkClusterJobParser with SparkClusterProcessMonitor {
+
+  import SparkClusterProcess._
+  import spray.httpx.SprayJsonSupport._
+  import SparkClusterJsonProtocol._
+
+  implicit lazy val ec: ExecutionContext = system.dispatcher
+  lazy val completionPromise = Promise[TerminalStatus]()
+  lazy val monitorPromise = Promise[Unit]()
+  val tag = this.getClass.getSimpleName
+  lazy val logger = Logger(LoggerFactory.getLogger(getClass.getName))
+
+  override def sendAndReceive: SendReceive = sendReceive
+
+  override def startMonitoringSparkClusterJob(jobPath: Path,  jsonFile: String): Future[TerminalStatus] = {
+    Future(parseJsonForSubmissionIdAndStatus(jobPath.resolve(jsonFile))) onComplete  {
+      case Success(resp: SparkJobSubmissionResponse) =>
+        val rcPath = jobPath.resolve("rc")
+        monitorSparkClusterJob(resp.submissionId, rcPath, monitorPromise)
+        evaluateMonitoringFuture(rcPath)
+      case Failure(exception: Throwable) =>
+        logger.error(s"{} Spark Job failed to submit successfully following reason: {}", tag, exception.getMessage)
+        completionPromise success Failed(exception)
+    }
+    completionPromise.future
+  }
+
+  override def monitorSparkClusterJob(subId: String,  rcPath: Path, promise: Promise[Unit]): Unit = {
+    pollForJobStatus(subId) onComplete {
```

Comment:
You need to provide a fall through Success() state in the match. I realize that it's not practically a thing but the compiler can't enforce that due to the guard statements.